ht.el

```diff
@@ -162,6 +162,14 @@ For every key-value pair in TABLE, evaluate FORM with the
 variables key and value bound."
   `(ht-each (lambda (key value) ,form) ,table))
 
+(defun ht-select-keys (table keys)
+  "Return a new TABLE according to specific KEYS."
+  (let (result)
+    (setq result (make-hash-table))
+    (dolist (key keys result)
+      (if (gethash key table)
```

Comment:
This isn't quite right: if `nil` is the value in `table`, it won't be set in `result`. You could filter using `ht-items`, or use `(gethash key table 'not-found-sentinel-value)`.